OSDOCS#14032:  Update the z-stream RN for 4.18.9 (#92209)
openshift/openshift-docs: release_notes/ocp-4-18-release-notes.adoc

* Previously, the Secret Janitor was not properly scoped in the `hypershift-operator`, which caused improper secret cleanup. This resulted in token secrets accumulating over time, disrupting the secret management process, while using annotation scoping with two instances of the `hypershift-operator`. With this release, a fix ensures that the secret cleanup continues as expected in a Red Hat OpenShift Kubernetes Service (ROKS) cluster that is managed by the `hypershift-operator`. The large amount of token secrets is eliminated and the proper secret management is maintained. (link:https://issues.redhat.com/browse/OCPBUGS-54498[*OCPBUGS-54498*])

* Previously, a bug occurred due to the incorrect handling of the etcd URL, preventing access to the `Kyverno` service. This resulted in DNS errors during the `kyverno` validation, preventing users on a {product-title} cluster with {hcp} from creating additional test groups. With this release, users can create additional test groups without encountering DNS errors during the `kyverno` validation. (link:https://issues.redhat.com/browse/OCPBUGS-54411[*OCPBUGS-54411*])

> * Previously, insufficient permissions led to the persistence of the `disk.csi.azure.com/agent-not-ready=value:NoExecute` taint after creating new {azure-first} disk Container Storage Interface (CSI) driver nodes. With this release, a fix disables the removal of the `not-ready` taint for {azure-short} disk CSI driver nodes, making the scheduler adhere to the `volume-attach-limit`. (link:https://issues.redhat.com/browse/OCPBUGS-54383[*OCPBUGS-54383*])

Reviewer: ```suggestion
* Previously, insufficient permissions led to the persistence of the `disk.csi.azure.com/agent-not-ready=value:NoExecute` taint after creating {azure-first} disk Container Storage Interface (CSI) driver nodes. With this release, a fix disables the removal of the `not-ready` taint for {azure-short} disk CSI driver nodes, making the scheduler adhere to the `volume-attach-limit` value. (link:https://issues.redhat.com/browse/OCPBUGS-54383[*OCPBUGS-54383*])
```

> * Previously, containers that used the SELinux domain of `container_logreader_t` to view container logs on a host at `/var/log` could not access logs in the `/var/log/containers` subdirectory. This was because of a missing symbolic link. With this release, a symbolic link is created for `/var/log/containers` so the containers can access the logs in `/var/log/containers`. (link:https://issues.redhat.com/browse/OCPBUGS-54342[*OCPBUGS-54342*])

Reviewer: ```suggestion
* Previously, containers that used the SELinux domain of `container_logreader_t` to view container logs on a host in the `/var/log` directory could not access logs in the `/var/log/containers` subdirectory. This was because of a missing symbolic link. With this release, a symbolic link is created so the containers can access the logs in the `/var/log/containers` subdirectory. (link:https://issues.redhat.com/browse/OCPBUGS-54342[*OCPBUGS-54342*])
```

> * Previously, an image pull secret that was generated for the internal Image Registry was not regenerated until after the embedded credentials expired This resulted in the image pull secrets being temporarily invalid. With this release, a fix for the image pull secrets are refreshed before the embedded credentials expire. (link:https://issues.redhat.com/browse/OCPBUGS-54304[*OCPBUGS-54304*])

Reviewer: ```suggestion
* Previously, an image pull secret that was generated for the internal Image Registry was not regenerated until after the embedded credentials expired. This resulted in the image pull secrets being temporarily invalid. With this release, the image pull secrets are refreshed before the embedded credentials expire. (link:https://issues.redhat.com/browse/OCPBUGS-54304[*OCPBUGS-54304*])
```

* Previously, the cluster autoscaler stopped scaling because of a failed machine in a machine set. This condition occurred due to inaccuracies in the way the cluster autoscaler counted machines in various non-running phases. With this release, the inaccuracies are fixed, allowing the cluster autoscaler to have an exact count. (link:https://issues.redhat.com/browse/OCPBUGS-53241[*OCPBUGS-53241*])

> * Previously, the Time-Group Management (T-GM) status was incorrectly announced as `S2` before the Digital Phase Locked Loop (DPLL) was locked during the Global Navigation Satellite System (GNSS) holdover. This caused inaccurate synchronization. With this release, the `DPLL` state decision logic is modified to ensure that the T-GM-STATUS moves to `S2` only after both phase offsets are valid and DPLL is in "Locked Holdover Acquired" state. This software update guarantees that the T-GM status accurately reflects the DPLL state when the GNSS source starts. (link:https://issues.redhat.com/browse/OCPBUGS-52956[*OCPBUGS-52956*])

Reviewer: ```suggestion
* Previously, the telecom grandmaster (T-GM) status was incorrectly announced as `S2` before the digital phase-locked loop (DPLL) was locked during the global navigation satellite system (GNSS) holdover. This caused inaccurate synchronization. With this release, the `DPLL` state decision logic is modified to ensure that the T-GM status moves to `S2` only after both phase offsets are valid and DPLL is in "Locked Holdover Acquired" state. This guarantees that the T-GM status accurately reflects the DPLL state when the GNSS source starts. (link:https://issues.redhat.com/browse/OCPBUGS-52956[*OCPBUGS-52956*])
```

> * Previously, a User-Defined Network (UDN) pod that is not qualified to use `egressIP` had to use its own UDN pod IP address instead of its node IP address as the source IP address in egressing packets. With this release, the UDN pod network is advertised correctly. (link:https://issues.redhat.com/browse/OCPBUGS-50965[*OCPBUGS-50965*])

Reviewer: ```suggestion
* Previously, a User-Defined Network (UDN) pod that is not qualified to use egress IP address had to use its own UDN pod IP address instead of its node IP address as the source IP address in egressing packets. With this release, the UDN pod network is advertised correctly. (link:https://issues.redhat.com/browse/OCPBUGS-50965[*OCPBUGS-50965*])
```

* Previously, when the cluster certificate authority (CA) bundle was updated with a custom CA bundle, a delay occurred for the change to reflect in the `insights-runtime-extractor` container. This issue occurred if the Insights Operator gathered data after the CA bundle was updated. With this release, a fix removes the delay and this issue no longer occurs. (link:https://issues.redhat.com/browse/OCPBUGS-48790[*OCPBUGS-48790*])

> * Previously, in {product-title} 4.17, a bug occurred because the code did not verify the load balancer IP address in the control plane subnet's Classless Inter-Domain Routing (CIDR) range. This resulted in the IP address existing outside of the valid range. This caused a 400 error during the installation. With this release, this fix prevents the 400 error because of  private IP address conflicts, ensuring a successful deployment of private {product-title} clusters on Azure. (link:https://issues.redhat.com/browse/OCPBUGS-43724[*OCPBUGS-43724*])

Reviewer: ```suggestion
* Previously, in {product-title} 4.17, the code did not verify the load balancer IP address in the control plane subnet Classless Inter-Domain Routing (CIDR) range. This resulted in the IP address existing outside of the valid range and caused a 400 error during the installation. With this release, this fix prevents the 400 error caused by private IP address conflicts, ensuring a successful deployment of private {product-title} clusters on {azure-short}. (link:https://issues.redhat.com/browse/OCPBUGS-43724[*OCPBUGS-43724*])
```
